clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -1037,7 +1038,7 @@ public void testHeartbeatRequestWithFencedInstanceIdException() throws Interrupt
             }
             fail("Expected pollHeartbeat to raise fenced instance id exception in 1 second");
         } catch (RuntimeException exception) {
-            assertTrue(exception instanceof FencedInstanceIdException);
+            assertInstanceOf(FencedInstanceIdException.class, exception);
```

Comment:
this should be rewrite by `assertThrow`

Reply:
Huge thanks for the detailed view again ! :smiley: 

> I have pushed a commit to rebase code and fix conflict. will merge it if there is no connected failure.

You are so kind! Next time I'll try to fix the conflict faster...